include/seqan3/alphabet/all.hpp

```diff
@@ -39,9 +39,121 @@
+ * for qualities, RNA structures and alignment gaps. In addition there are templates for combining alphabet
+ * type into new alphabets and wrappers for existing data types like the canonical `char`.
+ *
+ * These are defined in sub-modules of the alphabet module, the collaboration diagram at the top of this page
+ * provides an overview. Many provide features beyond that of the generic alphabet concept which we define below.
+ * The inheritance diagram of seqan3::alphabet_concept also provides a good overview of the more refined
+ * concepts and also the classes implementing them.
+ *
+ * ## The alphabet concept
+ *
+ * The seqan3::alphabet_concept defines the requirements a type needs to meet to be considered an alphabet
+ * by SeqAn, or in other words: you can expect certain properties and functions to be defined on
+ * all data types we call an alphabet.
+ *
+ * ### alphabet size
+ *
+ * All alphabets in SeqAn have a fixed size. It
+ * can be queried via the seqan3::alphabet_size metafunction and often also the `value_size` static
+ * member of the alphabet (see below for "free functions vs member functions").
+ *
+ * In some areas we provide alphabets types with different sizes for the same purpose, e.g. seqan3::dna4
+ * ('A', 'C', 'G', 'T'), seqan3::dna5 (plus 'N') and seqan3::dna15 (plus ambiguous characters defined by
+ * IUPAC). By convention most of our alphabets carry their size in their name (seqan3::dna4 has size 4 a.s.o.).
+ *
+ * A main reason for choosing a smaller alphabet over a bigger one is the possibility of **optimising for
+ * space efficiency**. Note, however, that a single letter by itself can never be smaller than a byte for
+ * architectural reasons. Actual space improvements are realised via secondary structures, e.g. when
+ * using a seqan3::bitcompressed_vector<seqan3::dna4> instead of std::vector<seqan3::dna4>. Also a
+ * a single letter composition of seqan3::dna4 with a quality alphabet may well fit into one byte, whereas
+ * a composition of seqan3::dna15 with a quality alphabet will definitely require two bytes per letter.
+ *
+ * ### assigning and retrieving values
+ *
+ * As mentioned above, we typically think of alphabets in their character representation, but we also
+ * require them in "rank representation" as programmers. In C and C++ it is quite difficult to
+ * cleanly differentiate between these, because the `char` type is considered an integral type and can
+ * be used to index an array (e.g. <tt>my_array['A']</tt> is valid). Moreover the sign of `char` is
+ * implementation defined and the smallest integer types `int8_t`, `uint8_t` are literally the
+ * same types as `signed char` and `unsigned char` on many platforms.
+ *
+ * This leads to ambiguity when assigning and retrieving values:
+ *
+ * ```cpp
+ * // WRONG EXAMPLE:
+ * dna4 my_letter{0};      // we want to set the default, an A
+ * dna4 my_letter{'A'};    // we also want to set an A, but we are setting value 65
```

Comment:
I'd rather give it a little more 'use' to the use case. Nobody will want to initialize values just like that.. (at least me :smile: ). It might be clearer with this:

```cpp
// Example
std::vector alphabet_count = {0, 0, 0, 0};

for (dna4 const & letter : some_dna4string)
    ++alphabet_count[letter]; 
// in the above line we want 'A','C','G','T' to be converted to ranks 0,1,2,3 
// instead of their ascii representation 65,67,71,84. But how do we know what happens? 

dna4 max_letter{get_position_of_max(alphabet_count)}; // e.g. max_letter{3}
std::cout << max_letter; // you expect 'T', but can we always be sure? 
```